x-pack/plugins/ml/public/application/settings/calendars/edit/calendar_form/dst_event_generator.tsx

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type * as estypes from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import type { EuiComboBoxOptionOption } from '@elastic/eui';
+import { EuiButton, EuiComboBox, EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import type { FC } from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
+import moment from 'moment-timezone';
+import { createDstEvents } from '../../dst_utils';
+
+interface Props {
+  addEvents: (events: estypes.MlCalendarEvent[]) => void;
+  setTimezone: (timezone: string) => void;
+}
+
+export const DstEventGenerator: FC<Props> = ({ addEvents, setTimezone }) => {
+  const generateEvents = () => {
+    if (selectedTimeZones.length === 0) {
+      return;
+    }
+    const events = createDstEvents(selectedTimeZones[0].value!);
+    addEvents(events);
+  };
+  const [selectedTimeZones, setSelectedTimeZones] = useState<
+    Array<EuiComboBoxOptionOption<string>>
+  >([]);
+
+  useEffect(() => {
+    if (selectedTimeZones.length > 0) {
+      setTimezone(selectedTimeZones[0].value!);
+    }
+  }, [selectedTimeZones, setTimezone]);
+
+  const timeZoneOptions = useMemo(() => {
+    return moment.tz.names().map((tz) => {
+      return {
+        label: tz,
+        value: tz,
+      };
+    });
+  }, []);
+
+  return (
+    <>
+      <EuiFlexGroup wrap gutterSize="s">
+        <EuiFlexItem grow={false} css={{ width: '400px' }}>
+          <EuiComboBox
+            placeholder="Select time zone"
+            singleSelection={{ asPlainText: true }}
+            options={timeZoneOptions}
+            selectedOptions={selectedTimeZones}
+            onChange={setSelectedTimeZones}
+          />
+        </EuiFlexItem>
+        <EuiFlexItem grow={false}>
+          <EuiButton
+            key="ml_import_event"
+            data-test-subj="mlCalendarDstGenerateEventsButton"
+            iconType="importAction"
+            onClick={generateEvents}
```

Comment:
For some calendars e.g. `Etc/GMT`, this action does nothing, presumably because these time zones not based on locations don't have the concept of DST? What about adding some indication to the user that there are no DST events for this calendar, otherwise it's not clear if the action just failed.

Reply:
I've added a info callout for timezone which do not generate DST events.